Terminal.Gui/Input/Keyboard/KeyBindings.cs

```diff
@@ -0,0 +1,279 @@
+    ///         Adds a new key combination that will trigger the commands in <paramref name="commands"/> on the View
+    ///         specified by <paramref name="target"/>.
+    ///     </para>
+    ///     <para>
+    ///         If the key is already bound to a different array of <see cref="Command"/>s it will be rebound
+    ///         <paramref name="commands"/>.
+    ///     </para>
+    /// </summary>
+    /// <remarks>
+    /// </remarks>
+    /// <param name="key">The key to check.</param>
+    /// <param name="target">The View the commands will be invoked on. If <see langword="null"/>, the key will be bound to <see cref="Application"/>.</param>
+    /// <param name="commands">
+    ///     The command to invoked on the <see paramref="target"/> when <paramref name="key"/> is pressed. When
+    ///     multiple commands are provided,they will be applied in sequence. The bound <paramref name="key"/> strike will be
+    ///     consumed if any took effect. 
+    /// </param>
+    public void Add (Key key, View? target, params Command [] commands)
+    {
+        KeyBinding binding = new (commands, target);
+        Add (key, binding);
+    }
+
+    private readonly Dictionary<Key, KeyBinding> _bindings = new (new KeyEqualityComparer ());
+
+    /// <summary>
+    ///     Gets the bindings.
+    /// </summary>
+    /// <returns></returns>
+    public IEnumerable<KeyValuePair<Key, KeyBinding>> GetBindings ()
+    {
+        return _bindings;
+    }
+
+    /// <summary>
+    ///     Gets the keys that are bound.
+    /// </summary>
+    /// <returns></returns>
+    public IEnumerable<Key> GetBoundKeys () { return _bindings.Keys; }
+
+    /// <summary>
+    ///     The view that the <see cref="KeyBindings"/> are bound to.
+    /// </summary>
+    /// <remarks>
+    ///     If <see langword="null"/> the KeyBindings object is being used for Application.KeyBindings.
+    /// </remarks>
+    public View? Target { get; init; }
+
+    /// <summary>Removes all <see cref="KeyBinding"/> objects from the collection.</summary>
+    public void Clear () { _bindings.Clear (); }
+
+    /// <summary>
+    ///     Removes all key bindings that trigger the given command set. Views can have multiple different keys bound to
+    ///     the same command sets and this method will clear all of them.
+    /// </summary>
+    /// <param name="command"></param>
+    public void Clear (params Command [] command)
+    {
+        KeyValuePair<Key, KeyBinding> [] kvps = _bindings
+                                                .Where (kvp => kvp.Value.Commands.SequenceEqual (command))
+                                                .ToArray ();
+
+        foreach (KeyValuePair<Key, KeyBinding> kvp in kvps)
+        {
+            Remove (kvp.Key);
+        }
+    }
+
+    /// <summary>Gets the <see cref="KeyBinding"/> for the specified <see cref="Key"/>.</summary>
+    /// <param name="key"></param>
+    /// <returns></returns>
+    public KeyBinding Get (Key key)
```

Comment:
`Get` has a generic meaning and `GetKey` or `GetKeyFromKey` may be more appropriate, like bellow you have `GetCommands` for e.g..